### ISSUE TITLE
Legic Prime support

### ISSUE THREAD

**Issue by Builderhummel** [flipperdevices/flipperzero-firmware#1688]
### Description of the feature you're suggesting.

Legic Prime is a standard similar to Mifare and it is common in europe.

It would be helpful to be able to read and emulate the legic prime cards just like mifare.

(The algorithm is also available in the Proxmark3 source code, like the Mifare one)

### Anything else?

The flipper project is really interesting. Legic Prime support would make it a lot more practical.

**Comment by david-bla** [comment]
Legic Prime is really common here in europe.
There is a big market for cheap access systems and the fact that it is totally reversed (broken) does not mean security consultants don't recommend it anymore.
This would be a great addition for the flipper :+1: .

I am not sure but seems like the hardware should be capable of it.
For reference see the implementation of proxmark3 like @Builderhummel mentioned:

https://github.com/Proxmark/proxmark3/blob/master/client/cmdhflegic.c

and its implementations:

https://github.com/Proxmark/proxmark3/blob/master/armsrc/legicrf.c
https://github.com/Proxmark/proxmark3/blob/master/common/legic_prng.c

**Comment by BuckarooBanzay** [comment]
can't really help on the implementing front here but i have the hardware to test it, just ping me up

**Comment by skotopes** [comment]
We'll come back to this after RFID rewriting in C

**Comment by Builderhummel** [comment]
@skotopes Thanks. The Frequency tag is wrong tho, since Legic Prime uses 13,56 MHz.

**Comment by panki27** [comment]
Worth noting that the Proxmark sources use its built-in FPGA, so adapting the code is going to be challenging.

Here is a talk about how the Legic protocol was reversed: https://fahrplan.events.ccc.de/congress/2009/Fahrplan/events/3709.en.html
The linked slides have a lot of detail.

**Comment by Tetris81** [comment]
is there any progress?

**Comment by Tetris81** [comment]
@skotopes is there any progress?

**Comment by skotopes** [comment]
@DrZlo13 wanna take a look?

**Comment by skotopes** [comment]
> @skotopes is there any progress?

Can you attach RFID RAW records here?

**Comment by Tetris81** [comment]
@skotopes i have one but is my work card
But I found a patch for LegicPrime on the site, maybe that will help:

https://itooktheredpill.irgendwo.org/2010/mensakarte-entschluesselt/
https://itooktheredpill.irgendwo.org/2010/legic-karten-beschreiben/  "Den eigentlichen Patch gibt es hier: legicfastreader.diff"
>> legicfastreader.diff 
[legicfastreader.diff.txt](https://github.com/flipperdevices/flipperzero-firmware/files/10610861/legicfastreader.diff.txt)

I don't know anything about programming but it looks like c+

**Comment by skotopes** [comment]
We still need some keys or records. Is there any place where we can order it online?

**Comment by Philipp2683** [comment]
@skotopes Do you have discord?

**Comment by panki27** [comment]
@skotopes One of their obscurity strategies is afaik to (almost) only sell B2B. But after some searching, I found this shop in the UK that doesn't require you to request a quote:
https://www.windmill-computing.com/products/view-one/pn/BDG-ISO-LEGIC-P1024/page/smart-cards

**Comment by skotopes** [comment]
@panki27 @Astrrra will research this protocol when she get time

**Comment by Astrrra** [comment]
HI! Support for Legic Prime is technically possible for us, we will implement it in the future.

**Comment by achtnullzwei** [comment]
@Astrrra I suspect I have some legic cards and I did a raw read from rfid menu. Lmk if the files can be of help. Ability to read/emulate those would be awesome.

**Comment by parsley** [comment]
Hi!
It looks like this FR belongs to this forum post:
https://forum.flipperzero.one/t/legic-prime/14159
Right?

**Comment by devBioS** [comment]
Legic Prime is technically all about timing, there are good resources about problems with timing and their solutions while emulating tags on the proxmark forum and their source.
The tag raw values are "encrypted" with a XOR value also present on each card.
Most importantly they use custom non-standard crc8 and crc16 generators to validate values present on the card which need to be calculated on the fly if emulating cards.

Also here the proxmark forum is a very good source of knowledge for prime tags..
Raw support for Legic Prime tags would be really awesome, where one could build an app upon :)

**Comment by Boringsapiens** [comment]
Hey @gornekich @skotopes any updates?

**Comment by skotopes** [comment]
Not really

**Comment by marvinto23** [comment]
Any update?

**Comment by skotopes** [comment]
Nope, it is in the queue, but with low priority. There is no specific date that I can promise to you guys. Really sorry.

**Comment by Nomcon** [comment]
In case the RFID raw values are still relevant.
I just captured the ones of my Legic chip.
I had to rename them to tip them so in order to upload, do with that what you will.
[RfidRecord.zip](https://github.com/user-attachments/files/18264028/RfidRecord.zip)

**Comment by fr33z3** [comment]
Any update here?

**Comment by derDeno** [comment]
How is it possible that such an important feature (that even seems very doable) is more than 3 years later still low priority? It's like one of the main reasons for many why they even bought their flipper zero.

**Comment by swissbyte** [comment]
Hi there is there any progress? Can we be of help to make this festure happen? I would be also happy to donate money for this feature.

Thanks
